### PR TITLE
feat: add limit and offset for filtering

### PR DESCRIPTION
**Pull request by numb3r3**

### CODE REVIEW

File: annlite/index.py [hunk 1]
```diff
@@ -378,24 +384,74 @@ def _search_documents(
...
-    def _cell_selection(self, query_np, limit):
+    def get_doc_by_id(self, doc_id: str):
+        """Get the document by id.
+
+        :param doc_id: the document id.
+        """
+
+        return self._get_doc_by_id(doc_id)
+
+    def get_docs(
```
Comment (jemmyshin) [inline annlite/index.py:428]: I prefer `det_docs_by_filtering`

File: README.md [hunk 2]
```diff
@@ -123,6 +123,28 @@ Or shorten the loop as one-liner using the element & attribute selector:
...
+And you can also get the documents with `limit` and `offset`, which is useful for pagination:
+
+```python
+docs = ann.get_docs(limit=10, offset=0)
+```
+
+Furthermore, you can also get the documents ordered by a specific column from the index:
+
+```python
+docs = ann.get_docs(limit=10, offset=0, order_by='x', ascending=True)
```
Comment (jemmyshin) [inline README.md:143]: we should tell users to put `x` into `filterable_attrs` when initializing `annlite` if they want to use this feature

File: annlite/container.py [hunk 3]
```diff
@@ -148,43 +148,55 @@ def filter_cells(
         cells: 'np.ndarray',
         where_clause: str = '',
         where_params: Tuple = (),
-        limit: int = 10,
+        limit: int = -1,
```
[the remark this answers is not in the document]
Comment (numb3r3) [reply annlite/container.py:151]: return all of the elements if given `-1`

File: annlite/index.py (hunk 1 above)
Comment (numb3r3) [reply annlite/index.py:428]: we already have an API `def filter()` which works for filtering (in this API, the `filter` parameter is not optional anymore).

File: README.md (hunk 2 above)
Comment (numb3r3) [reply README.md:143]: I have added the note message below.

**Review by jemmyshin — APPROVED** [review]
LGTM

**Review by OrangeSodahub — APPROVED** [review]
LGTM

**Pull request merged**